components/BackToTop/BackToTopButton.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useEffect, useState } from "react";
+import { FaArrowUp } from "react-icons/fa";
+import { SCROLL_LIMIT } from "../../app/constants";
+
+export const BackToTopButton = () => {
+  const [scrollY, setScrollY] = useState(0);
+
+  useEffect(() => {
+    const handleScroll = () => {
+      setScrollY(window.scrollY);
+    };
+
+    handleScroll();
+
+    window.addEventListener("scroll", handleScroll);
+
+    return () => {
+      window.removeEventListener("scroll", handleScroll);
+    };
+
+  }, []);
+
+  const handleClick = () => {
+    window.scrollTo({
+      top: 0,
+      behavior: "smooth",
+    })
+  }
+
+  if (scrollY > SCROLL_LIMIT) {
+    return (
+      <div className="group fixed bottom-12 right-12">
+        <button 
+          className="focus:animate-button-press rounded-2xl border border-gray-100 bg-violet-600 p-4 text-white shadow-xl focus:ring group-hover:border-dashed group-hover:border-violet-400 group-hover:bg-white dark:drop-shadow-[5px_5px_8px_rgba(124,58,237,0.25)] dark:group-hover:bg-[#101623]"
```

Comment:
```suggestion
          className="focus:animate-button-press rounded-full bg-violet-600 p-4 text-white shadow-xl focus:ring group-hover:border-dashed group-hover:border-violet-400 group-hover:bg-white dark:drop-shadow-[5px_5px_8px_rgba(124,58,237,0.25)] dark:group-hover:bg-[#101623]"
```

Reply:
That's why I added the `border border-gray-100`, but `border border-gray-100 md:border-violet-600` would be better. The border is the same color like the background from the `md` breakpoint.

![back-to-top-btn](https://user-images.githubusercontent.com/41576384/216759131-f802a4d2-a0b0-4f3a-8c26-ab0b1502b5ba.png)

Reply:
Oh, yeah thats fine,but make it based on the theme

Reply:
Looks perfect, thank you